ui/apps/platform/src/Containers/MainPage/Body.tsx

```diff
@@ -86,45 +88,34 @@ function Body(): ReactElement {
         >
             <ErrorBoundary>
                 <Switch>
-                    <ProtectedRoute path={dashboardPath} component={AsyncDashboardPage} />
-                    <ProtectedRoute path={networkPath} component={AsyncNetworkPage} />
-                    <ProtectedRoute path={violationsPath} component={AsyncViolationsPage} />
-                    <ProtectedRoute path={compliancePath} component={AsyncCompliancePage} />
-                    <ProtectedRoute path={integrationsPath} component={AsyncIntegrationsPage} />
-                    <ProtectedRoute path={policiesPath} component={AsyncPoliciesPagePatternFly} />
-                    <ProtectedRoute path={riskPath} component={AsyncRiskPage} />
-                    <ProtectedRoute
-                        path={accessControlPathV2}
-                        component={AsyncAccessControlPageV2}
-                    />
-                    <ProtectedRoute path={apidocsPath} component={AsyncApiDocsPage} />
-                    <ProtectedRoute path={userBasePath} component={AsyncUserPage} />
-                    <ProtectedRoute path={systemConfigPath} component={AsyncSystemConfigPage} />
-                    <ProtectedRoute
-                        path={vulnManagementReportsPath}
-                        component={AsyncVulnMgmtReports}
-                        featureFlagEnabled={isVulnReportingEnabled}
-                        requiredPermission="VulnerabilityReports"
-                    />
-                    <ProtectedRoute
+                    <Route path={dashboardPath} component={AsyncDashboardPage} />
+                    <Route path={networkPath} component={AsyncNetworkPage} />
+                    <Route path={violationsPath} component={AsyncViolationsPage} />
+                    <Route path={compliancePath} component={AsyncCompliancePage} />
+                    <Route path={integrationsPath} component={AsyncIntegrationsPage} />
+                    <Route path={policiesPath} component={AsyncPoliciesPagePatternFly} />
+                    <Route path={riskPath} component={AsyncRiskPage} />
+                    <Route path={accessControlPathV2} component={AsyncAccessControlPageV2} />
+                    <Route path={apidocsPath} component={AsyncApiDocsPage} />
+                    <Route path={userBasePath} component={AsyncUserPage} />
+                    <Route path={systemConfigPath} component={AsyncSystemConfigPage} />
+                    {isVulnReportingEnabled && hasVulnerabilityReportsPermission && (
```

Comment:
Aha, there is a subtle difference depending on whether first or second level route:

1. /main/system-health-pf without feature flag
    * original code redirects to /main/dashboard because it matches but `ProtectedRoute` redirects
    * refactored code redirects to /main/dashboard because no match falls through to `<Redirect from={mainPath} to={dashboardPath} />`
2. /main/vulnerability-management/reports simulate without feature flag or permission
    * original code redirects to /main/dashboard because it matches but `ProtectedRoute` redirects
    * refactored code does not match the full route, but does match /main/vulnerability-management partial route, which renders `NotFoundMessage` element, although not because of permissions:
    ![Reporting](https://user-images.githubusercontent.com/11862657/161820956-73ce66ea-5456-4e5e-816a-082d32c95de7.png)
 
Comments about the plus and minus of the difference in behavior are welcome.

> communicate to the user that they don't have permission to view the page (but the page exists) versus that they visited a non-existent URL

@vjwilson Can you remind us the plus and minus of how much to communicate versus conceal?

Reply:
The recommended way to deal with not having enough permissions to visit a URL is to show some kind of "404" of page, like the one I attached is. (Redirecting to dashboard is acceptable from a security perspective.)

Anything that says "this is here, but you don't have permission to see it" is bad. (I know we do that for Access Control, but we eventually need to fix that.)

Reply:
> The Reporting item is not in the left nav, so I had to type in the URL.
> I got our "Not Found" page, which is probably the correct thing to show because a security rule of thumb is not to reveal that something exists if a user does not have permission to see it.

I was wondering if the fact that we show a `Not Found` page vs. redirect can imply something is potentially there 👀. Should we then not redirect but show a `Not Found` page in general? The message would differ in that:

1. If you have no permissions to view, it'll say you don't have permission to view the page
2. If the feature flag is protected, you'll just see that what you're looking for can't be found
3. If it's not a valid route, you'll just see that what you're looking for can't be found

Reply:
^ I guess Van's point makes sense. I'm fine with keeping the redirect or keeping things consistent and just showing a `Not Found` page in general

Reply:
This thread has good discussion.
1. Having made recommended changes, I am merging to give these changes a full week before end of sprint.
2. Let's confirm on Thursday the potential change from main dashboard to not found in time for a small follow up.